modules/abstract-utxo/src/parseOutput.ts

```diff
@@ -228,14 +228,20 @@ export async function parseOutput({
      * details are fetched on the wallet, a local address validation is run, whose errors however are generated
      * client-side and can therefore be analyzed with more granularity and type checking.
      */
-    const addressDetails = await fetchAddressDetails({
-      reqId,
-      addressDetailsVerification,
-      addressDetailsPrebuild,
-      currentAddress,
-      disableNetworking,
-      wallet,
-    });
+    const isCurrentAddressInRecipients = txParams.recipients?.some((recipient) =>
+      recipient.address.includes(currentAddress)
+    );
+    // Retrieve address details if the address is not part of the recipients list
+    const addressDetails = isCurrentAddressInRecipients
+      ? undefined
+      : await fetchAddressDetails({
+          reqId,
+          addressDetailsVerification,
+          addressDetailsPrebuild,
+          currentAddress,
+          disableNetworking,
+          wallet,
+        });
     // verify that the address is on the wallet. verifyAddress throws if
     // it fails to correctly rederive the address, meaning it's external
     currentAddressType = AbstractUtxoCoin.inferAddressType(addressDetails) || undefined;
```

Comment:
I'm surprised this compiles - the argument to `inferAddressType` does not allow `undefined` here. This is probably because `fetchAddressDetails` returns `any`.

I'm pretty sure that `inferAddressType(undefined)` fails and we end up in that overly broad `catch` block.

Reply:
https://github.com/BitGo/BitGoJS/pull/3440#discussion_r1157514684

Reply:
please rewrite the code so that we return early and not in the `catch` block

Reply:
done